src/s1ap/handlers/s1ap_msg_delegator.c

```diff
@@ -244,28 +298,33 @@ parse_nas_pdu(char *msg,  int nas_msg_len, struct nasPDU *nas,
 				nas->elements[2].esm_info_tx_required = true;
 			}
 		}
-		offset += len;
-		}
+		msg += len;
 
 		/*DRX parameter*/
-		offset += 4;
+		msg += 3;
 
 		/*MS network capability*/
-		nas->elements[4].ms_network.element_id = msg[offset];
-		++offset;
-		nas->elements[4].ms_network.len = msg[offset];
-		++offset;
-		memcpy((nas->elements[4].ms_network.capab), msg+offset,
+		nas->elements[4].ms_network.element_id = msg[0];
+		msg++;
+		nas->elements[4].ms_network.len = msg[0];
+		msg++;
+		memcpy(nas->elements[4].ms_network.capab, msg,
 			nas->elements[4].ms_network.len);
+		log_msg(LOG_INFO, "element_id=%x\n", nas->elements[4].ms_network.element_id);
+		log_msg(LOG_INFO, "len=%x\n", nas->elements[4].ms_network.len);
+		log_msg(LOG_INFO, "network.capab=%s\n", msg_to_hex_str((char *)nas->elements[4].ms_network.capab, nas->elements[4].ms_network.len, &buffer));
+		free(buffer);
```

Comment:
In case of element not being received, the code crashes at this free. Should have a check to see if the buffer is allocated or do the allocation and free inside the msg_to_hex_str function

Reply:
@badhrinathpa feel free (no pun intended :-)) to fix the issue as you seem fit, the main purpose of this kind of logging was to observe payload during debug. The following is probably safer
``` if (ptr != NULL)   free(ptr);```
I do not think free inside the msg_to_hex_str function is an option, as we need to log outside the function.

Reply:
that's a good one.. :) 
will approve your fix and then update this fix as part of some other merge.